torch/distributed/_tensor/sharding_prop.py

```diff
@@ -231,6 +231,8 @@ def spec_to_strategy(spec: object) -> object:
                         if output_strategy.input_specs is None
                         else output_strategy.input_specs[idx]
                     )
+                    if not desired_spec.tensor_meta:
```

Comment:
After a second thought, I think only fixing this is not enough. i.e. if there's only `output_spec`, then the `desired_spec` would be `output_spec` and it would be associated with every input arg metadatas (it's actually overriding), so it would make things wrong.

What I think we should do here:
1. we make PlacementStrategy to have `input_specs` as "non-optional", which makes each op should generate placement strategy with input specs, even if the input spec and output spec is the same
2. we audit our existing ops and change those ops who only have a output_spec to also include the input dtensor spec, where it's constructed separately with the output_spec with the same `mesh` and `placements`.

Reply:
If we make PlacementStrategy to have `input_specs` non-optional, what should we put in the field when using `spec_to_strategy` to generate initial input strategies?

Btw a somewhat hacky way that doesn't require big change might be associating `desired_spec` with a deep copy of `output_spec` if `input_specs is None`

Reply:
That's a good point actually.. Maybe we should simply auditing our existing ops and make sure the generated OpStrategy have input_specs, without forcing PlacementStrategy.input_specs be non-optional.

Then in the place we have here, we just simply:
1. force `desired_spec = output_strategy.input_specs[idx]`
2. do assertions. assert that there must have `desired_spec.tensor_meta`.